test/test_parser.py

```diff
@@ -81,7 +79,7 @@ def get_output(self):
             roots[key] = root
 
             for error in errors:
-                errors_str += f'{error.level.name}: {os.path.basename(error.filename)}:{error.line}: {error.message}\n'  # noqa: E501
+                errors_str += f'{error.level.name}: {error.get_message(basename=True)}\n'
```

Comment:
I'm a bit confused on this one, and bear in mind I'm going by the partial diffs and the commit messages only.

You mention a possibility of having filename being `None`, is that right? The `basename=True` path is not really about that though, it's about calling `basename` or not, but still using the original name, which apparently may be `None`. If it were about that, it could check for `None` without any additional argument like we do already with the `line`.

Reply:
Ah, I see. Actually it made sense now that I read it again. Yeah, feel free to disregard ;)